stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -356,22 +356,21 @@ private void batchProcessing(final TypedRecord<?> initialCommand) {
           recordProcessors.stream()
               .filter(p -> p.accepts(command.getValueType()))
               .findFirst()
-              .orElse(null);
-      if (currentProcessor != null) {
-        currentProcessingResult = currentProcessor.process(command, processingResultBuilder);
-
-        final BatchProcessingStepResult batchProcessingStepResult =
-            collectBatchProcessingStepResult(
-                currentProcessingResult,
-                lastProcessingResultSize,
-                // +1 since we already need include the current command in the calculation
-                pendingCommands.size() + processedCommandsCount + 1,
-                currentProcessingBatchLimit);
-
-        pendingCommands.addAll(batchProcessingStepResult.toProcess());
-        pendingWrites.addAll(batchProcessingStepResult.toWrite());
-        currentProcessingResult.getProcessingResponse().ifPresent(pendingResponses::add);
-      }
+              .orElseThrow(() -> new NoSuchProcessorException(command));
```

Comment:
❓ 🤔 Can this be an issue when we stop supporting a certain command, and write different command now ( I think this happened in the past). This would mean we have no longer a processor for an old command 🤔 Hm before we just skipped it? 🤔 I'm a bit unsure here.

Reply:
This would be a problem, yeah :+1:
But if we stop using a command, we can't just remove the processor for it because we need to process anything that's already on the logstream. So it would be correct to stop here, incorrect to just remove the processor.

Reply:
Hmm. Yes, we should not remove processors when we deprecate commands. While this may have happened in the past, we also keep such processors around. For example, we still support the deprecated deployment distributions while deployments are distributed with command distribution nowadays:

https://github.com/camunda/zeebe/blob/4e17f4375c77b5087b7d1ae447e19f7fff2d18f3/engine/src/main/java/io/camunda/zeebe/engine/processing/EngineProcessors.java#L253-L266

We could benefit from tests that deter us from removing processors.

Reply:
Alright, thanks for confirming :+1: 

I think this discussion is resolved then?